storage/filesystem/fat/tests/TESTS/fat/fat_filesystem/main.cpp

```diff
@@ -134,28 +134,28 @@ void test_read_dir()
     err = dir.open(&fs, "test_read_dir");
     TEST_ASSERT_EQUAL(0, err);
 
-    struct dirent *de;
+    struct dirent de;
     bool test_dir_found = false;
     bool test_file_found = true;
 
-    while ((de = readdir(&dir))) {
-        printf("d_name: %.32s, d_type: %x\n", de->d_name, de->d_type);
+    while (dir.read(&de)) {
```

Comment:
Rechecking the documentation, this should apparently be testing `dir.read(&de) == 1`.   It returns 1 for success, 0 for end of directory and negative for error. This would infinite loop on error.

Reply:
No problem, thanks for catching that! I should've double checked it before making the PR.

Reply:
Done